src/reader/reader.ts

```diff
@@ -47,24 +49,153 @@ export function putContentIntoLines(contents: string): Query[] {
+  // You should get two matches
+  const match = regex.exec(content) || [];
+  if (match.length >= 2) {
+    // Replace everything between the delimters except newlines
+    // Note .* will not match newline without /s flag
+    const newlines = match[2].replace(/.*/g, "");
+    // Now replace whole regex with just the newlines left over
+    content = content.replace(regex, "$1" + newlines);
+  }
+  return content;
+}
+
+/**
+ * Strip comments as may contain SQL-like syntax which can confuse the lexer
+ * Preserve new lines so output still makes sense
  */
 function stripComments(content: string): string {
-  const contentInLines = content.split(Keyword.Newline);
-
-  for (let i = 0; i < contentInLines.length; i++) {
-    if (
-      contentInLines[i].startsWith(Keyword.CommentDash) ||
-      contentInLines[i].startsWith(Keyword.CommentHash) ||
-      contentInLines[i].startsWith(Keyword.CommentMultiLineStart)
-    ) {
-      contentInLines[i] = "";
-    }
+  // Add a new line to end in case not there to make regex matching easier
+  let added_new_line = false;
+  if (content[-1] != "\n") {
+    content = content + "\n";
+    added_new_line = true;
+  }
+
+  // Remove any content after a dash comment until end of line
+  const comment_dash_regex = new RegExp(Keyword.CommentDash + ".*?\n", "");
+  content = content.replace(comment_dash_regex, "\n");
+
+  // Remove any content after a hash comment until end of line
+  const comment_hash_regex = new RegExp(Keyword.CommentHash + ".*?\n", "");
+  content = content.replace(comment_hash_regex, "\n");
+
+  // Remove any content in a multi-line comment - but preserve new lines
+  content = multiLineReplace(
+    content,
+    "",
+    Keyword.CommentMultiLineStart,
+    Keyword.CommentMultiLineEnd
+  );
+
+  // Remove any new line we added
+  if (added_new_line) {
+    content = content.slice(0, -1);
+  }
+
+  return content;
+}
+
+/**
+ * Strip out anything of the format:
+ *   language js as $$ ... $$
+ * Even over multi-lines
+ * Note this depends on dotall (/./s) so requires Node 9 or greater
+ */
+function stripLanguages(content: string): string {
+  // The separator ($$) is above example can be different so find it
+  const start_language_regex = /language\s*\w*\s*as\s*([^\s]*)/gis;
+
+  // Iterate though all separators and remove the non-SQL code
+  let result;
+  while ((result = start_language_regex.exec(content)) !== null) {
+    const separator = result[1].trim();
+    content = multiLineReplace(
+      content,
+      "language\\s*\\w*\\s*as\\s*",
+      separator,
+      separator
+    );
+  }
+
+  return content;
+}
+
+/**
+ * Sanitise quotes to remove ; which confuses lexer
+ */
+function sanatiseQuotes(content: string): string {
+  let result;
+
+  // Then remove any ; from within single quotes
+  const single_quote_regex = /('[^\']*;[^\']*')/gi;
+  while ((result = single_quote_regex.exec(content)) !== null) {
+    const sanitised_result = result[0].replace(/;/g, "");
```

Comment:
Can we use `camelCase` instead of `snake_case` for variable naming too?

Reply:
Will do.

Reply:
Done.